beps/0005-definitive-plugin-listing/README.md

```diff
@@ -0,0 +1,240 @@
+know that this has succeeded?
+-->
+
+1. As an integrator, I can now get a list of currently installed features across my deployment.
+1. As an integrator/administrator, I can get a list of currently installed features for a single instance in my deployment.
+1. As an integrator/administrator, I can continue to use my static `HostDiscovery` config.
+
+### Non-Goals
+
+<!--
+What is out of scope for this BEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+1. The list of plugins will not change unless there is a config change, they will not be dynamically loaded.
+1. The list of plugins will not be used in routing or discovery.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation.
+-->
+
+Most of the work below is API definitions for hosting the list of features and better abstractions to make that list available through HTTP. This is not intended to perform much heavy lifting, that can happen in a further design. We will build a default implementation on the existing static `HostDiscovery` config definition for split deployments. Users can override the API as they need.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs or even code snippets.
+If there's any ambiguity about HOW your proposal will be implemented, this is the place to discuss them.
+-->
+
+### Glossary
+
+### New Type `BackendFeatureMeta`
+
+```ts
+type BackendFeatureMeta =
+  | {
+      type: 'plugin';
+      pluginId: string;
+    }
+  | {
+      type: 'module';
+      pluginId: string;
+      moduleId: string;
+    };
+```
+
+### New `InstanceMetadataService`
+
+While we could attach the existing information to the `PluginMetadataService`, we propose a new service that handles instance-level information. The existing `PluginMetadataService` should reveal information about the plugin itself, its `pluginId`, dependencies or similar. The new `InstanceMetadataService` should give you information about the entire Backstage instance that you're interrogating. One could imagine this service also having information about instance URLs, health or gateway status.
+
+```ts
+interface InstanceMetadataService {
+  getInstalledFeatures: () => BackendFeatureMeta[];
+}
+```
+
+Currently, this would need to be bootstrapped directly in `BackendInitializer` or `BackstageBackend`. It may make sense to create a new `FeatureRegistry` class that can expose this instead similar to the `PluginMetadataService`.
+
+### New HTTP API to expose `InstanceMetadataService`
+
+We also propose exposing the new services through an HTTP API, so that these can be queried.
+
+```yaml
+paths:
+  /.backstage/instanceInfo/features/installed:
```

Comment:
So to be clear, just features in this instance right?

Reply:
yes